From 094bf1721be9cdd59ce81453397e5ac1fe183cdc Mon Sep 17 00:00:00 2001
From: Brian Wood <brian.j.wood@intel.com>
Date: Thu, 18 Sep 2014 15:56:44 -0700
Subject: [PATCH] Add in Kconfig/Makefile changes for enabling BCMDHD

This commit adds in the changes for the Kconfig and Makefile
for enabling the Broadcom BCMDHD driver. Also, by default the
BCMDHD driver was configured to only allows building as built-in
I changing this to allow for building as a module.

Signed-off-by: Brian Wood <brian.j.wood@intel.com>
---
 drivers/net/wireless/Kconfig         |  1 +
 drivers/net/wireless/Makefile        |  1 +
 drivers/net/wireless/bcmdhd/Makefile | 20 ++++++++++++++------
 3 files changed, 16 insertions(+), 6 deletions(-)

diff --git a/drivers/net/wireless/Kconfig b/drivers/net/wireless/Kconfig
index eba19fe8eaf8..71748bfdd41d 100644
--- a/drivers/net/wireless/Kconfig
+++ b/drivers/net/wireless/Kconfig
@@ -273,6 +273,7 @@ source "drivers/net/wireless/ath/Kconfig"
 source "drivers/net/wireless/b43/Kconfig"
 source "drivers/net/wireless/b43legacy/Kconfig"
 source "drivers/net/wireless/brcm80211/Kconfig"
+source "drivers/net/wireless/bcmdhd/Kconfig"
 source "drivers/net/wireless/hostap/Kconfig"
 source "drivers/net/wireless/ipw2x00/Kconfig"
 source "drivers/net/wireless/iwlwifi/Kconfig"
diff --git a/drivers/net/wireless/Makefile b/drivers/net/wireless/Makefile
index 0fab227025be..826d63eded2e 100644
--- a/drivers/net/wireless/Makefile
+++ b/drivers/net/wireless/Makefile
@@ -57,5 +57,6 @@ obj-$(CONFIG_MWIFIEX)	+= mwifiex/
 
 obj-$(CONFIG_BRCMFMAC)	+= brcm80211/
 obj-$(CONFIG_BRCMSMAC)	+= brcm80211/
+obj-$(CONFIG_BCMDHD)	+= bcmdhd/
 
 obj-$(CONFIG_CW1200)	+= cw1200/
diff --git a/drivers/net/wireless/bcmdhd/Makefile b/drivers/net/wireless/bcmdhd/Makefile
index 4e468476c20e..f91199412a8c 100644
--- a/drivers/net/wireless/bcmdhd/Makefile
+++ b/drivers/net/wireless/bcmdhd/Makefile
@@ -107,13 +107,15 @@ DHDCFLAGS += -DCUSTOM_SET_SHORT_DWELL_TIME
 # m: module type driver
 # y: built-in type driver
 ##########################
-DRIVER_TYPE ?= y
+# Using defconfig CONFIG_BCMDHD to determine
+# if building a module or built-in, not local Makefile config
+#DRIVER_TYPE ?= y
 
 #########################
 # Chip dependent feature
 #########################
 
-ifneq ($(filter y, $(CONFIG_BCM4354) $(CONFIG_BCM4356)),)
+ifneq ($(filter y m, $(CONFIG_BCM4354) $(CONFIG_BCM4356)),)
   DHDCFLAGS += -DUSE_WL_TXBF
   DHDCFLAGS += -DUSE_WL_FRAMEBURST
   DHDCFLAGS += -DCUSTOM_DPC_CPUCORE=0
@@ -142,7 +144,10 @@ ifneq ($(CONFIG_BCMDHD_SDIO),)
   DHDCFLAGS += -DMAX_HDR_READ=128
   DHDCFLAGS += -DDHD_FIRSTREAD=128
 
-  DHDCFLAGS += -DSAR_SUPPORT
+# Disabling SAR (Specific Absorption Rate) testing API since Broadcom
+# doesn't provide necessary extern function defines with upstream
+# opensource driver
+#  DHDCFLAGS += -DSAR_SUPPORT
 endif
 
 ifneq ($(CONFIG_BCMDHD_PCIE),)
@@ -155,7 +160,10 @@ ifneq ($(CONFIG_BCMDHD_PCIE),)
 # Disable watchdog thread
   DHDCFLAGS += -DCUSTOM_DHD_WATCHDOG_MS=0
 
-  DHDCFLAGS += -DSAR_SUPPORT
+# Disabling SAR (Specific Absorption Rate) testing API since Broadcom
+# doesn't provide necessary extern function defines with upstream
+# opensource driver
+#  DHDCFLAGS += -DSAR_SUPPORT
 endif
 endif
 
@@ -185,7 +193,7 @@ endif
 
 #EXTRA_LDFLAGS += --strip-debug
 
-ifeq ($(DRIVER_TYPE),y)
+ifeq ($(CONFIG_BCMDHD),y)
   DHDCFLAGS += -DENABLE_INSMOD_NO_FW_LOAD
   DHDCFLAGS += -DUSE_LATE_INITCALL_SYNC
 endif
@@ -213,7 +221,7 @@ ifneq ($(CONFIG_BCMDHD_PCIE),)
 endif
 
 bcmdhd-objs := $(DHDOFILES)
-obj-$(DRIVER_TYPE)   += bcmdhd.o
+obj-$(CONFIG_BCMDHD)   += bcmdhd.o
 
 all:
 	@echo "$(MAKE) --no-print-directory -C $(KDIR) SUBDIRS=$(CURDIR) modules"
-- 
2.0.3

